### PR TITLE
JSDK 2675: regional edge hostnames

### PR DESCRIPTION
**Pull request by makarandp0**
update signaling and insights endpoints.
- no real change for signaling endpoint. Just need to test with new endpoints.
- for insights it will be:
`wss://sdkgw.us1.twilio.com/v1/VideoEvents` => `wss://${region}.vss.twilio.com/insights`

**Before merge**:
- [ ] add unit tests
- [ ] verify endpoints as they become available.

**Contributing to Twilio**

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [ ] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

File: lib/util/endpoints.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+'use strict';
+
+module.exports.getSignalingEndpoint = (environment, region) => {
```
Comment (manjeshbhargav) [inline lib/util/endpoints.js]: @makarandp0 We can refactor the VSG hostname calculation logic into its own function:
```js
function getVsgHostName(environment, region) {
  region = region === 'gll' ? 'global' : encodeURIComponent(region);
  return environment === 'prod'
    ? `wss://${region}.vss.twilio.com`
    : `wss://${region}.vss.${environment}.twilio.com`
}
```
Then, `getSignalingEndpoint` and `getInsightsEndpoint` will become:
```js
function getInsightsEndpoint(environment, region) {
  return `${getVsgHostName(environment, region)}/insights`;
}

function getSignalingEndpoint(environment, region) {
  return `${getVsgHostName(environment, region)}/signaling`;
}

exports.getInsightsEndpoint = getInsightsEndpoint;
exports.getSignalingEndpoint = getSignalingEndpoint;
```

File: lib/util/constants.js [hunk 2]
```diff
@@ -26,16 +22,18 @@ module.exports.PUBLISH_BACKOFF_MS = 20;
...
-      return [
-        { urls: 'stun:global.stun.twilio.com:3478?transport=udp' }
-      ];
-    default:
-      return [
-        { urls: `stun:global.stun.${environment}.twilio.com:3478?transport=udp` }
-      ];
-  }
+  // eslint-disable-next-line no-console
+  console.error('unexpected DEFAULT_ICE_SERVERS was referenced:', environment);
```
Comment (manjeshbhargav) [inline lib/util/constants.js]: @makarandp0 Is this temporary? Please remove this before merging.

File: lib/util/endpoints.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+}
+
+function getDefaultIceServers(environment) {
+  return environment === 'prod'
+    ? [{ urls: 'stun:global.stun.twilio.com:3478?transport=udp' }]
+    : [{ urls: `stun:global.stun.${environment}.twilio.com:3478?transport=udp` }];
+}
+
+function getVsgHostName(environment, region) {
+  region = region === 'gll' ? 'global' : encodeURIComponent(region);
```
Comment (syerrapragada) [inline lib/util/endpoints.js]: @makarandp0 should gll be translated to roaming?

**Comment by makarandp0** [comment]
These changes were incorporated in https://github.com/twilio/twilio-video.js/pull/895

**Pull request closed without merging**